pymeasure/instruments/instrument.py

```diff
@@ -191,103 +226,214 @@ class attributes that are a DynamicProperty. Check also for class variables
+                )
         return super().__getattribute__(name)
 
     @property
     def complete(self):
-        """ This property allows synchronization between a controller and a device. The Operation Complete
+        """This property allows synchronization between a controller and a device. The Operation Complete
         query places an ASCII character 1 into the device's Output Queue when all pending
         selected device operations have been finished.
         """
         if self.SCPI:
-            return self.ask("*OPC?").strip()
+            ready = self.ask_no_lock("*OPC?").strip()
+            if ready == "1":
+                return True
+            elif ready == "0":
+                return False
+            else:
+                return None
         else:
-            raise NotImplementedError("Non SCPI instruments require implementation in subclasses")
+            raise NotImplementedError(
+                "Non SCPI instruments require implementation in subclasses"
+            )
 
     @property
     def status(self):
-        """ Requests and returns the status byte and Master Summary Status bit. """
-        if self.SCPI:
-            return self.ask("*STB?").strip()
+        """Checks the status of the system.
+
+        Returns:
+            "ok: busy" if the threading lock cannot be acquired, "ok: ON"
+            if the system is on and the lock was acquired. If communication
+            was not initially successfull, the system is queried again and
+            the ID is checked. If the system does not return an expected response,
+            status is set to warning: Cannot communicate with device".
+        """
+        if not self.communication_success:
+            id = self.get_id(check_for_errors=False)
+
+            if id.startswith(self.id_starts_with):
+                curr_status = "ok: ON"
+                self.communication_success = True
+            else:
+                curr_status = "warning: Cannot communicate with device"
         else:
-            raise NotImplementedError("Non SCPI instruments require implementation in subclasses")
+            if self._lock.locked():
+                curr_status = "ok: busy"
+            else:
+                curr_status = "ok: ON"
+
+        return curr_status
 
     @property
     def options(self):
-        """ Requests and returns the device options installed. """
+        """Requests and returns the device options installed."""
         if self.SCPI:
             return self.ask("*OPT?").strip()
         else:
-            raise NotImplementedError("Non SCPI instruments require implementation in subclasses")
+            raise NotImplementedError(
+                "Non SCPI instruments require implementation in subclasses"
+            )
 
-    @property
-    def id(self):
-        """ Requests and returns the identification of the instrument. """
+    def get_id(self, check_errs=True):
```

Comment:
Why not be as verbose here as in `ask`?
```suggestion
    def get_id(self, check_for_errors=True):
```